cpp/lib/include/ZoteroHarvesterConfig.h

```diff
@@ -0,0 +1,214 @@
+/** \brief Classes related to the Zotero Harvester's configuration data
+ *  \author Madeeswaran Kannan
+ *
+ *  \copyright 2019 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+
+#include <memory>
+#include <set>
+#include <unordered_map>
+#include "IniFile.h"
+#include "RegexMatcher.h"
+
+
+namespace ZoteroHarvester {
+
+
+// This namespace contains classes that represent the (immutable) configuration
+// data of the Zotero Harvester program. Global, group and journal parameters
+// read in from an INI file. Refer to the documentation in the default configuration
+// INI file for details about individual configuration fields/keys.
+namespace Config {
+
+
+// Parameters that pertain to all harvestable journals/groups.
+struct GlobalParams {
+    enum INIKey : unsigned {
+        TRANSLATION_SERVER_URL,
+        ENHANCEMENT_MAPS_DIRECTORY,
+        GROUP_NAMES,
+        STRPTIME_FORMAT_STRING,
+        SKIP_ONLINE_FIRST_ARTICLES_UNCONDITIONALLY,
+        DOWNLOAD_DELAY_DEFAULT,
+        DOWNLOAD_DELAY_MAX,
+        RSS_HARVEST_INTERVAL,
+        RSS_FORCE_PROCESS_FEEDS_WITH_NO_PUB_DATES,
+        TIMEOUT_CRAWL_OPERATION,
+        TIMEOUT_DOWNLOAD_REQUEST
+    };
+
+    std::string translation_server_url_;
+    std::string enhancement_maps_directory_;
+    std::string group_names_;
+    std::string strptime_format_string_;
+    bool skip_online_first_articles_unconditonally_;
+    struct {
+        unsigned default_delay_;
```

Comment:
This is documented in the actual .conf file, as mentioned in the top-most comment of this namespace. Wanted to avoid having multiple places where the unit is specified (in case it should ever be changed in the future).